File: 1008.cc
nclude <stdio.h>
#include <string.h>
#include <iostream>
using namespace std;

int ncase;
/*19 months. Each of the first 18 months was 20 days long
pop, no, zip, zotz, tzec, xul, yoxkin, mol, chen,
yax, zac, ceh, mac, kankin, muan, pax, koyab, cumhu*/

/*Tzolkin 
The year was divided into 13 periods, each 20 days long
20 names: imix, ik, akbal, kan, chicchan, cimi, manik, lamat, muluk, ok, 
chuen, eb, ben, ix, mem, cib, caban, eznab, canac, ahau
*/
int daysOfTheMonth(char monthName[]){
    if (strcmp(monthName,"uayet") ==0)
    {
        return 5;
    }
    return 20;
}

int indexOfTheMonth(char monthName[]){
    if (strcmp(monthName,"pop") ==0) return 0;
    if (strcmp(monthName,"no") ==0) return 1;
    if (strcmp(monthName,"zip") ==0) return 2;
    if (strcmp(monthName,"zotz") ==0) return 3;
    if (strcmp(monthName,"tzec") ==0) return 4;
    if (strcmp(monthName,"xul") ==0) return 5;
    if (strcmp(monthName,"yoxkin") ==0) return 6;
    if (strcmp(monthName,"mol") ==0) return 7;
    if (strcmp(monthName,"chen") ==0) return 8;
    if (strcmp(monthName,"yax") ==0) return 9;
    if (strcmp(monthName,"zac") ==0) return 10;
    if (strcmp(monthName,"ceh") ==0) return 11;
    if (strcmp(monthName,"mac") ==0) return 12;
    if (strcmp(monthName,"kankin") ==0) return 13;
    if (strcmp(monthName,"muan") ==0) return 14;
    if (strcmp(monthName,"pax") ==0) return 15;
    if (strcmp(monthName,"koyab") ==0) return 16;
    if (strcmp(monthName,"cumhu") ==0) return 17;
    if (strcmp(monthName,"uayet") ==0) return 18;
}

char * indexTostring(int index){
    char* name[20]={ "imix", "ik", "akbal", "kan", 
        "chicchan", "cimi", "manik", "lamat", "muluk", "ok", 
        "chuen", "eb", "ben", "ix", "mem", "cib", "caban", "eznab", 
        "canac", "ahau"};
    return name[index];
}
int main(){
    scanf("%d", &ncase);
    int i;
    printf("%d\n",ncase);
    for (i= 0 ;i<ncase;i++)
    {
        int numberoftheday, year;
        char month[10];
        scanf("%d.%s%d", &numberoftheday, month, &year);

        //cout<<"The numberoftheday is "<<numberoftheday<<endl;
        //cout<<"The month is "<<month<<endl;
        //cout<<"The year is "<<year<<endl;
        //puts("before die");
        long int sumdays = numberoftheday +1 + indexOfTheMonth(month)*20+ year*365;


        long int anotherYear = (long int) sumdays / 260;
        long int remainderDays = sumdays % 260;
        if (!remainderDays)
        {
            anotherYear --;
        }
        long int printDay = remainderDays % 13;
        long int stringIndex = remainderDays % 20;
        //cout<<"The month index is "<<stringIndex<<endl;
        if (stringIndex == 0)
        {
            stringIndex = 20;
            
        }
        if (printDay == 0)
        {
            printDay = 13;
        }
        printf("%d %s %d\n",printDay, indexTostring(stringIndex-1), anotherYear);
        //cout<<"Output day is "<< printDay<<endl;
        //cout<<"Month name is "<< indexTostring(stringIndex-1)<<endl;
        //cout<<"Another year is "<<anotherYear <<endl;


    }

}

File: 1008.java
 import java.util.Scanner;

 public class Main {

     public static void main(String[] args) {
         Scanner scan = new Scanner(System.in);
         int times = scan.nextInt();
         System.out.println(times);
         int day;
         int day2;
         int t_day = 0;
         int t_month;
         String month;
         String month2;
         int year;
         int year2;
         int day_num;
         int day_num2;
         for (int i = 0; i < times; i++) {
             String str_day = scan.next();
             day = Integer.parseInt(str_day.substring(0, str_day.length() - 1));
             month = scan.next();
             year = scan.nextInt();
             t_month = getHaabMonth(month);
             if(t_month<=18){
                 t_day = (t_month-1) * 20;
             }else{
                 t_day = 360;
             }
             day_num = 365 * year  +  t_day + (day + 1);
             year2 = day_num / 260;
             day_num2 = day_num % 260;
             if(day_num2 == 0){
                 year2--;
             }
             day2 = day_num2 % 13;
             if (day2 == 0) {
                 day2 = 13;
             }
             month2 = getTzolkinMonth(day_num2 % 20);
             System.out.println(day2 + " " + month2 + " " + year2);
         }
     }

     public static int getHaabMonth(String month) {
         if (month.equals("pop")) {
             return 1;
         } else if (month.equals("no")) {
             return 2;
         } else if (month.equals("zip")) {
             return 3;
         } else if (month.equals("zotz")) {
             return 4;
         } else if (month.equals("tzec")) {
             return 5;
         } else if (month.equals("xul")) {
             return 6;
         } else if (month.equals("yoxkin")) {
             return 7;
         } else if (month.equals("mol")) {
             return 8;
         } else if (month.equals("chen")) {
             return 9;
         } else if (month.equals("yax")) {
             return 10;
         } else if (month.equals("zac")) {
             return 11;
         } else if (month.equals("ceh")) {
             return 12;
         } else if (month.equals("mac")) {
             return 13;
         } else if (month.equals("kankin")) {
             return 14;
         } else if (month.equals("muan")) {
             return 15;
         } else if (month.equals("pax")) {
             return 16;
         } else if (month.equals("koyab")) {
             return 17;
         } else if (month.equals("cumhu")) {
             return 18;
         } else if (month.equals("uayet")) {
             return 19;
         } else {
             return 0;
         }
     }

     public static String getTzolkinMonth(int month) {
         if (month == 1) {
             return "imix";
         } else if (month == 2) {
             return "ik";
         } else if (month == 3) {
             return "akbal";
         } else if (month == 4) {
             return "kan";
         } else if (month == 5) {
             return "chicchan";
         } else if (month == 6) {
             return "cimi";
         } else if (month == 7) {
             return "manik";
         } else if (month == 8) {
             return "lamat";
         } else if (month == 9) {
             return "muluk";
         } else if (month == 10) {
             return "ok";
         } else if (month == 11) {
             return "chuen";
         } else if (month == 12) {
             return "eb";
         } else if (month == 13) {
             return "ben";
         } else if (month == 14) {
             return "ix";
         } else if (month == 15) {
             return "mem";
         } else if (month == 16) {
             return "cib";
         } else if (month == 17) {
             return "caban";
         } else if (month == 18) {
             return "eznab";
         } else if (month == 19) {
             return "canac";
         } else if (month == 0) {
             return "ahau";
         } else {
             return "";
         }
     } 
}